assemblies/assembly-optaplanner-school-timetable-quarkus.adoc

```diff
@@ -0,0 +1,58 @@
+[id='assembly-optaplanner-school-timetable-quarkus']
+= Creating a school timetable using {PLANNER} with Quarkus
+
+ifdef::context[:parent-context: {context}]
+:context: optaplanner-quarkus
+
+This guide walks you through the process of creating a Quakus
+application with {PLANNER}'s constraint solving artificial intelligence (AI). You will build a REST application that optimizes a school timetable for students and teachers.
+
+image::optimizer/timeTableAppScreenshot.png[]
+
+
+Your service will assign `Lesson` instances to `Timeslot` and `Room` instances automatically
+by using AI to adhere to the following hard and soft _scheduling constraints_:
+
+* A room can have at most one lesson at the same time.
+* A teacher can teach at most one lesson at the same time.
+* A student can attend at most one lesson at the same time.
+* A teacher prefers to teach in a single room.
+* A teacher prefers to teach sequential lessons and dislikes gaps between lessons.
+
+Mathematically speaking, school timetabling is an _NP-hard_ problem.
+That means it is difficult to scale.
+Simply iterating through all possible combinations with brute force would take millions of years
+for a non-trivial dataset, even on a supercomputer.
+Fortunately, AI constraint solvers such as {PLANNER} have advanced algorithms
+that deliver a near-optimal solution in a reasonable amount of time. What is considered to be a reasonable amount of time is subjective and depends on the goals of your problem.
+
+.Prerequisites
+
+* OpenJDK 8 or later is installed. Red Hat build of Open JDK is available from the https://access.redhat.com/jbossnetwork/restricted/listSoftware.html[Software Downloads] page in the Red Hat Customer Portal (login required).
```

Comment:
Starting from OptaPlanner 8, OpenJDK 11 is the minimum.
The only reason it works on OpenJDK 8 (despite docs etc saying it's 11+) is because quarkus 1.x reuses some of our jars to run platform tests. As soon as quarkus 2.x is out, we'll target 11 in the optaplanner 8.x not just officially but also in the jars.